skops/io/tests/test_persist.py

```diff
@@ -130,6 +132,16 @@ def _tested_estimators(type_filter=None):
     for name, Estimator in all_estimators(type_filter=type_filter):
         if Estimator in UNSUPPORTED_TYPES:
             continue
+
+        # Skip SGD-based estimators for sklearn > 1.5
+        if Version(sklearn_version) > Version("1.5") and name in {
+            "SGDClassifier",
+            "SGDRegressor",
+            "SGDOneClassSVM",
+            "PassiveAggressiveClassifier",
+            "GraphicalLassoCV",
+        }:
```

Comment:
I added this for lack of better ideas. the estimators above have a sgd loss, which we don't have in the trusted losses in the 1.6 version so the tests fail with them. what better can be done here?

Reply:
we should add those new objects to trusted sets instead.

Reply:
unfortunately they are still the old objects inporting from _sgd_fast. this most likely has something to do with how this part of the code is done in scikit-learn.

Reply:
@adrinjalali I was referring to this message as the last issue to address and to "undo the temp fix to see the error for yourself. It seemed counter intuitive to add the old loss to the `trusted` again."

Reply:
Not really, we add them to trsuted, with a commend that they're removed now, Their main module is still sklearn

Reply:
I need some more practical insight here. When we have a 1.6 estimator, we can't point back to the old `_sgd_fast` objects as they don't exist, and yet the `SGD` estimators in `all_estimators` still have those when retrieved. How does one do this? It must be the case that this was also an issue internally in `scikit-learn`, is there maybe an example?

Reply:
why would they have it? trusted values are only to allow loading objects w/o user having to specify trusted objects. Otherwise they have no impact on the loaded objects

Reply:
I believe I might not be understanding something in the skops design. Below is the failing test output. `sklearn.linear_model._sgd_fast.Hinge` doesn't exist in sklearn1.6, what should I do in this case? 

```

_______________________________________________________________________ test_can_persist_fitted[SGDOneClassSVM(max_iter=5)] _______________________________________________________________________

estimator = SGDOneClassSVM(max_iter=5, random_state=0)

    @pytest.mark.parametrize(
        "estimator", _tested_estimators(), ids=_get_check_estimator_ids
    )
    def test_can_persist_fitted(estimator):
        """Check that fitted estimators can be persisted and return the right results."""
        set_random_state(estimator, random_state=0)

        X, y = get_input(estimator)
        if get_tags(estimator).requires_fit:
            with warnings.catch_warnings():
                warnings.filterwarnings("ignore", module="sklearn")
                if y is not None:
                    estimator.fit(X, y)
                else:
                    estimator.fit(X)

        # test that we can get a list of untrusted types. This is a smoke test
        # to make sure there are no errors running this method.
        # it is in this test to save time, as it requires a fitted estimator.
        dumped = dumps(estimator)
        untrusted_types = get_untrusted_types(data=dumped)

        loaded = loads(dumped, trusted=untrusted_types)
>       assert_params_equal(estimator.__dict__, loaded.__dict__)

X          = array([[6.99380484e-01, 9.64606424e-01, 2.12523045e-01, 5.98946831e-02,
        7.62114512e-01, 4.57627354e-01, 1.0122...02249730e-01, 2.24258934e-01, 6.76460732e-01,
        1.35126740e+00, 1.31908640e-01, 3.75147117e-01, 1.50699840e+00]])
dumped     = b'PK\x03\x04\x14\x00\x00\x00\x00\x00\xaa\x82yYL>s\xdc \x01\x00\x00 \x01\x00\x00\x0e\x00\x00\x006066774608.npy\x93NUMPY...\x00\x80\x01\x00\x02\x00\x00schema.jsonPK\x05\x06\x00\x00\x00\x00\x03\x00\x03\x00\xb1\x00\x00\x00\xa4+\x00\x00\x00\x00'
estimator  = SGDOneClassSVM(max_iter=5, random_state=0)
loaded     = SGDOneClassSVM(max_iter=5, random_state=0)
untrusted_types = ['sklearn.linear_model._sgd_fast.Hinge']
y          = array([1, 1, 1, 0, 0, 1, 0, 1, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 0, 0, 0,
       0, 1, 0, 1, 0, 1, 0, 1, 0, 0, 1, 1, 0, 1, 0, 1, 0, 0, 0, 1, 1, 0,
       1, 1, 1, 1, 0, 0])

skops/io/tests/test_persist.py:398:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
skops/io/tests/_utils.py:174: in assert_params_equal
    _assert_vals_equal(val1, val2)
        key        = '_loss_function_'
        params1    = {'C': 1.0, '_loss_function_': <sklearn.linear_model._sgd_fast.Hinge object at 0x1699cad70>, 'alpha': 0.0001, 'average': False, ...}
        params2    = {'C': 1.0, '_loss_function_': <sklearn.linear_model._sgd_fast.Hinge object at 0x1699c8750>, 'alpha': 0.0001, 'average': False, ...}
        val1       = <sklearn.linear_model._sgd_fast.Hinge object at 0x1699cad70>
        val2       = <sklearn.linear_model._sgd_fast.Hinge object at 0x1699c8750>
skops/io/tests/_utils.py:131: in _assert_vals_equal
    _assert_generic_objects_equal(val1, val2)
        val1       = <sklearn.linear_model._sgd_fast.Hinge object at 0x1699cad70>
        val2       = <sklearn.linear_model._sgd_fast.Hinge object at 0x1699c8750>
skops/io/tests/_utils.py:65: in _assert_generic_objects_equal
    _assert_tuples_equal(val1.__reduce__(), val2.__reduce__())
        _is_builtin = <function _assert_generic_objects_equal.<locals>._is_builtin at 0x15eace050>
        val1       = <sklearn.linear_model._sgd_fast.Hinge object at 0x1699cad70>
        val2       = <sklearn.linear_model._sgd_fast.Hinge object at 0x1699c8750>
skops/io/tests/_utils.py:71: in _assert_tuples_equal
    _assert_vals_equal(subval1, subval2)
        subval1    = (1.0,)
        subval2    = (0.0,)
        val1       = (<class 'sklearn.linear_model._sgd_fast.Hinge'>, (1.0,))
        val2       = (<class 'sklearn.linear_model._sgd_fast.Hinge'>, (0.0,))
skops/io/tests/_utils.py:116: in _assert_vals_equal
    _assert_tuples_equal(val1, val2)
        val1       = (1.0,)
        val2       = (0.0,)
skops/io/tests/_utils.py:71: in _assert_tuples_equal
    _assert_vals_equal(subval1, subval2)
        subval1    = 1.0
        subval2    = 0.0
        val1       = (1.0,)
```
        val2       = (0.0,)

Reply:
(working on it locally)

so this is actually showing that we're doing something wrong. And investigating it, I see that not all loss functions are removed/moved. Only a few are removed and the rest are still there. I'm working on a fix.